common/trinary/ptrit.h

```diff
@@ -9,15 +9,199 @@
+
+#elif defined(PTRIT_SSE)
+#if !defined(__SSE__)
+#error __SSE__ is not defined
+#endif
+#include <immintrin.h>
+typedef __m128d ptrit_s;
+#define PTRIT_SIZE 128
+
+#else
+#error Invalid PTRIT_PLATFORM.
+
+#endif
+
+#if defined(PTRIT_CVT_ORN)
+#error Intel intrinsics curl impl must use PTRIT_CVT_ANDN
+#endif
+#if !defined(PTRIT_CVT_ANDN)
+// -1 -> (1,0); 0 -> (1,1); +1 -> (0,1); (0,0) -- NaT
+#define PTRIT_CVT_ANDN
+#endif
+
+#endif
 
 typedef struct {
   ptrit_s low;
   ptrit_s high;
 } ptrit_t;
 
+#if defined(PTRIT_NEON)
+#define ORN(x, y) vornq_u64(y, x)
+#define XOR(x, y) veorq_u64(x, y)
+#define AND(x, y) vandq_u64(x, y)
+#define OR(x, y) vorrq_u64(x, y)
+#define NOT(x) ORN(x, XOR(x, x))
+
+#define XORORN(x, y, z) XOR(x, ORN(y, z))
+#define ORORN(x, y, z) OR(x, ORN(y, z))
+
+#else
+#if defined(PTRIT_AVX512F)
+/*
+  x y z x^(~y&z)  x^(y&z) (x&y)&z
+  0 0 0   0         0         0
+  0 0 1   1         0         0
+  0 1 0   0         0         0
+  0 1 1   0         1         0
+  1 0 0   1         1         0
+  1 0 1   0         1         0
+  1 1 0   1         1         0
+  1 1 1   1         0         1
+         D2        78         80
+*/
+#define XORANDN(x, y, z) _mm512_ternarylogic_epi64(x, y, z, 0xD2)
+#define XORAND(x, y, z) _mm512_ternarylogic_epi64(x, y, z, 0x78)
+#define ANDAND(x, y, z) _mm512_ternarylogic_epi64(x, y, z, 0x80)
+#else
+
+#if defined(PTRIT_AVX2)
+#define ANDN(x, y) _mm256_andnot_si256(x, y)
+#define AND(x, y) _mm256_and_si256(x, y)
+#define XOR(x, y) _mm256_xor_si256(x, y)
+#define OR(x, y) _mm256_or_si256(x, y)
+#define NOT(x) _mm256_andnot_si256(x, _mm256_set_epi64x(-1ll, -1ll, -1ll, -1ll))
+
+#elif defined(PTRIT_AVX)
+#define ANDN(x, y) _mm256_andnot_pd(x, y)
+#define AND(x, y) _mm256_and_pd(x, y)
+#define XOR(x, y) _mm256_xor_pd(x, y)
+
+#elif defined(PTRIT_SSE2)
+#if defined(_MSC_VER) && _MSC_VER < 1900 /*&& !defined(_WIN64)*/
+static __inline __m128i _mm_set_epi64x(__int64 _I1, __int64 _I0) {
+  __m128i i;
+  i.m128i_i64[0] = _I0;
+  i.m128i_i64[1] = _I1;
+  return i;
+}
+#endif
+#define ANDN(x, y) _mm_andnot_si128(x, y)
+#define AND(x, y) _mm_and_si128(x, y)
+#define XOR(x, y) _mm_xor_si128(x, y)
+#define OR(x, y) _mm_or_si128(x, y)
+#define NOT(x) _mm_andnot_si128(x, _mm_set_epi64x(-1ll, -1ll))
+
+#elif defined(PTRIT_SSE)
+#define ANDN(x, y) _mm_andnot_pd(x, y)
+#define AND(x, y) _mm_and_pd(x, y)
+#define XOR(x, y) _mm_xor_pd(x, y)
+
+#else
+#define AND(x, y) ((x) & (y))
+#define XOR(x, y) ((x) ^ (y))
+#define OR(x, y) ((x) | (y))
+#define NOT(x) (~(x))
+#define ANDN(x, y) AND(NOT(x), (y))
+#define ORN(x, y) OR(NOT(x), (y))
+#endif  // PTRIT_PLATFORM generic
+
+#define XORANDN(x, y, z) XOR(x, ANDN(y, z))
+#define ANDAND(x, y, z) AND(x, AND(y, z))
+#define ORORN(x, y, z) OR(x, ORN(y, z))
+#define XORAND(x, y, z) XOR(x, AND(y, z))
+#define XORORN(x, y, z) XOR(x, ORN(y, z))
+#endif  // PTRIT_AVX512F
+#endif  // PTRIT_NEON
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// The lowest 2 bits of `t` are used.
```

Comment:
Can you use this kind of documentation? 
e.g:

/**
 * Add a trusted channel id into the api's trusted channels set
 *
 * @param api - The API [in,out]
 * @param pk - A new public key [in]
 *
 * @return return code
 */